src/dialog-editor/services/modal/modalService.ts

```diff
@@ -81,6 +85,40 @@ class ModalController {
             }
           );
         }
+        // set modal title
+        if (!this.modalData.dynamic) {
+          var titleLabel;
+
+          switch (this.modalData.type) {
```

Comment:
Personally I don't like switches. They are too big, hard to read and hard to interpret. I prefer something which is (at least for me) better to read.
```Typescript
const titles = {
  DialogFieldTextBox:         __('Text Box'),
  DialogFieldTextAreaBox:     __('Text Area'),
  DialogFieldCheckBox:        __('Check Box'),
  DialogFieldDropDownList:    __('Dropdown'),
  DialogFieldRadioButton:     __('Radio Button'),
  DialogFieldDateControl:     __('Datepicker'),
  DialogFieldDateTimeControl: __('Timepicker'),
  DialogFieldTagControl:      __('Tag Control')
}
const titleLabel = this.modalData.type in titles && titles[this.modalData.type];
this.modalTitle =  __(`Edit ${titleLabel} Field`);
```

Reply:
Hi @karelhala, there are 3 other ```switch``` statements in this file, so I think in this case it's more consistent to use a ```switch```

Reply:
@karelhala i like the changes you've suggested, but it's out of scope for this PR, I'll create a new PR with changes you've suggested.